module/VuFind/src/VuFind/DigitalContent/OverdriveConnector.php

```diff
@@ -0,0 +1,819 @@
+            $authorizationData = $patronTokenData->token_type . ' ' . $patronTokenData->access_token;
+            $headers = array(
+                "Authorization: $authorizationData",
+                "User-Agent: VuFind-Plus",
+                "Content-Type: application/json"
+            );
+            curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+            curl_setopt($ch, CURLOPT_FOLLOWLOCATION, 1);
+            curl_setopt($ch, CURLOPT_TIMEOUT, 30);
+            curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+            if ($requestType != null) {
+                curl_setopt($ch, CURLOPT_CUSTOMREQUEST, $requestType);
+            }
+            if ($params != null) {
+                curl_setopt($ch, CURLOPT_POST, 1);
+                //Convert post fields to json
+                $jsonData = array('fields' => array());
+                foreach ($params as $key => $value) {
+                    $jsonData['fields'][] = array(
+                        'name' => $key,
+                        'value' => $value
+                    );
+                }
+                $postData = json_encode($jsonData);
+                curl_setopt($ch, CURLOPT_POSTFIELDS, $postData);
+            } else {
+                curl_setopt($ch, CURLOPT_HTTPGET, true);
+            }
+            curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
+            
+            $return = curl_exec($ch);
+            $curlInfo = curl_getinfo($ch);
+            curl_close($ch);
+            
+            $this->debug("curl Info: ".print_r($curlInfo, true));
+            //if all goes well for DELETE, the code will be 204 and response is empty.
+            if ($requestType=="DELETE" &&  $curlInfo['http_code'] == 204) {
+                $this->debug("DELETE Patron call appears to have worked.");
+                return true;
+            }
+            
+            $returnVal = json_decode($return);
+            $this->debug("response from call: ".print_r($returnVal, true));
+            if ($returnVal != null) {
+                if (!isset($returnVal->message) 
+                    || $returnVal->message != 'An unexpected error has occurred.') {
+                    return $returnVal;
+                }
+            } else {
+                return $result;
+            }
+        } else {
+            $this->Error("not connected to Patron API");
+        }
+        return false;
+    }
+}
+
+/*
+Placeholder for ID prefix stuff. not sure if I need this
+
+    private function _stripPrefixes($ids)
+    {
+        $result = array();
+        foreach ($ids as &$id) {
+            $result[]= $this->_stripPrefix($id);
+        }
+        return $result;
+    }
+    private function _stripPrefix($id)
+    {
+        //TODO SET prefix in CONFIG
+        $prefix = 'overdrive.';
+        return substr($id, strlen($prefix));
+    }
+*/
+
+/**
+ * Class OverdriveResult
+ *
+ * @package VuFind\DigitalContent
+ */
+class OverdriveResult
```

Comment:
This class should probably get its own file for stylistic reasons.

Reply:
I'll probably just use stdClass instead.

Reply:
Sounds reasonable to me!